src/hooks/stations.ts

```diff
@@ -48,10 +48,26 @@ export const useBooking = () => {
         }),
     [BUCHUNGEN],
   );
+  const refreshBooking = useCallback(
+    () =>
+      getCurrentBooking()
+        .then(setBooking)
+        .then(() => cancelCurrentBooking())
```

Comment:
This function should probably handle the case when the booking has run out and the app hasn't noticed that yet. In this case it can happen that the station becomes empty or all remaining bikes are already booked.

Reply:
You can also use `async` functions here (for the complicated logic), no worries. I just didn't use them most of the time because they aren't compatible with `useEffect`.

Reply:
See 1215004d33f2e2ab8e1d355d66f1cf9aed4de75c.